File: src/main/java/groupwork/Movie.java
package groupwork;

public class Movie extends DomainObject {
    public static final int  CHILDRENS = 2;
    public static final int  REGULAR = 0;
    public static final int  NEW_RELEASE = 1;


    private int _priceCode;

    public Movie(String name, int priceCode) {
        _name = name;
        _priceCode = priceCode;
    }

    public int priceCode() {
        return _priceCode;
    }

    public void persist() {
        Registrar.add ("Movies", this);
    }

    public static Movie get(String name) {
        return (Movie) Registrar.get ("Movies", name);
    }
}

File: src/main/scala/groupwork/refactor/Movie.scala
package groupwork.refactor

import scala.util.Try

case class Movie(name: String, priceCode: MoviePriceCodes) extends DomainObject

object Movie {
  private val category = "Movies"
  
  def persist(movie: Movie): Try[Unit] = Registrar.add(category, movie)
  
  def get(name: String): Option[Movie] = Registrar.get(category, name) match {
    case result: Movie => Some(result)
    case _ => None
  }
}
